modules/schedule/index.js

```diff
@@ -0,0 +1,396 @@
+var later  = require('later');
+var moment = require('moment');
+var sugar  = require('sugar');
+var hash = require('json-hash'); 
+
+require("moment-duration-format");
+
+var bot;
+
+var minimumCreationDelay = 0; 
+var minimumInterval = 60; 
+var noCommands = false;
+var digestLength = 8;
+
+// digest -> schedule
+var schedules = {};
+var lastSchedule = null;
+// { 'id': <digest of data before adding timer and stuff>,
+//   'created': <created timestamp>,
+//   'schedule:' <later schedule>,
+//   'blame': <owner>,
+//   'calls': <number of calls left>,
+//   'channel': <channel to print in>,
+//   'command': <what to say / execute>,
+//   'timer': <our timer instance if any>
+//
+
+// m1 - m2 so pass in backwards for positive
+function getDifference(m1,m2) {
+  return moment.duration(moment(m1).diff(moment(m2), "milliseconds"));
+}
+
+function getAverageInterval(s) {
+  var numSamples = 5;
+  var next = later.schedule(s).next(numSamples+2); // skip NEXT occurrence
+  var totalSeconds = 0;
+
+  next.forEach(function(e,i,d) {
+    e = moment(e);
+  });
+  
+  for(i=1; i<=numSamples; i++)
+  {
+    totalSeconds += parseInt(getDifference(next[i+1], next[i]).format("s"));
+  }
+
+  return totalSeconds/numSamples;
+}
+
+function writeSchedule(data) {
+  bot.writeDataFile("later.json", 
+      JSON.stringify(schedules), function(err) {
+        if(err) console.log("Error writing command file: " + err);
+  });
+}
+
+function newSchedule(data) {
```

Comment:
TODO, use a real object so you can call 'new Schedule' instead of this and have things be more idiomatic and stuff.